From d81e470bc3d7647d85d2456885ebc86a5e19bc1f Mon Sep 17 00:00:00 2001
From: Trent Lloyd <trent@lloyd.id.au>
Date: Tue, 31 Mar 2015 09:00:09 +0800
Subject: [PATCH] avahi_server_set_browse_domains: check the provided domain
 list, instead of the list from the configuration file

(cherry-picked from commit 1616785d583939aa4c1b859fcf626759214769dd)

Related: #1340837
---
 avahi-core/server.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/avahi-core/server.c b/avahi-core/server.c
index 4acf5fc..dd9da99 100644
--- a/avahi-core/server.c
+++ b/avahi-core/server.c
@@ -1792,7 +1792,7 @@ int avahi_server_set_browse_domains(AvahiServer *s, AvahiStringList *domains) {
 
     assert(s);
 
-    for (l = s->config.browse_domains; l; l = l->next)
+    for (l = domains; l; l = l->next)
         if (!avahi_is_valid_domain_name((char*) l->text))
             return avahi_server_set_errno(s, AVAHI_ERR_INVALID_DOMAIN_NAME);
 
-- 
2.7.4

